File: 백준/Gold/2437. 저울/저울.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int N = Integer.parseInt(br.readLine());
        int[] weights = new int[N];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            weights[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(weights);

        int min = 1;

        for (int i = 0; i < N; i++) {

            if (min < weights[i]) {
                break;
            } else {
                min += weights[i];
            }
        }

        System.out.println(min);
    }
}

File: 백준/Gold/2437. 저울/저울.py
# 저울
import sys
input = sys.stdin.readline

N = int(input())
weights = list(map(int,input().split()))

weights.sort()

minn = 1

for i in range(N):

    if minn < weights[i]:
        break

    minn += weights[i]

print(minn)
     
